### PR TITLE
Fix Make generate  

### PR DESCRIPTION
**Pull request by ArangoGutierrez**

### CODE REVIEW

File: api/nfd/v1alpha1/types.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ type NodeFeatureRuleList struct {
 }
 
 // NodeFeatureRule resource specifies a configuration for feature-based
-// customization of node objects, such as node labeling.
+// customization of node objects, such as node labeling. LOOSEME
```
Comment (marquiz) [inline api/nfd/v1alpha1/types.go:115]: LOOSEME?

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,4 @@ demo/helper-scripts/*.pdf
 demo/helper-scripts/*.log
 /kustomization.yaml
 coverage.out
+vendor/
```
Comment (marquiz) [inline .gitignore:6]: I think it's better not to have this in gitignore as we don't ship with the vendored deps. With this change an out-of-date vendor/ directory get's ignored and easily causes unexpected build results

File: go.mod [hunk 3]
```diff
@@ -3,206 +3,203 @@ module sigs.k8s.io/node-feature-discovery
 go 1.22
 
 require (
+	cloud.google.com/go/compute v1.23.3
+	cloud.google.com/go/compute/metadata v0.2.3
+	github.com/Azure/azure-sdk-for-go v68.0.0+incompatible
```
Comment (marquiz) [inline go.mod:8]: We shouldn't depend on this stuff directly. I don't believe any changes to go.mod would be needed in this PR

File: hack/update_codegen.sh [hunk 4]
```diff
@@ -18,51 +18,56 @@ set -o errexit
...
+  rm -rf ${TMP_VENDOR_DIR}
+  rm sigs.k8s.io
+}
+
+trap cleanup EXIT
 GO_CMD=${1:-go}
 NFD_ROOT=$(realpath $(dirname ${BASH_SOURCE[0]})/..)
 
-go mod vendor
+"${GO_CMD}" mod vendor
```
Comment (marquiz) [inline hack/update_codegen.sh:33]: Remind me: we need the `vendor/` for running `go generate`? And thus can't just do `go mod vendor -o "$TMP_VENDOR"`?

File: api/nfd/v1alpha1/types.go (hunk 1 above)
Comment (ArangoGutierrez) [reply api/nfd/v1alpha1/types.go:115]: ups that was my hack to check things were being updated properly

File: .gitignore (hunk 2 above)
Comment (ArangoGutierrez) [reply .gitignore:6]: ok

File: hack/update_codegen.sh (hunk 4 above)
Comment (ArangoGutierrez) [reply hack/update_codegen.sh:33]: `"${GO_CMD}" generate ./cmd/... ./pkg/... ./source/...` fails if things are not in `vendor`

**Review by marquiz — APPROVED** [review]
/lgtm

**Pull request merged**